### ISSUE TITLE
About multiple sites and multiple DBNames

### ISSUE THREAD

**Issue by naretini** [laradock/laradock#417]
I see that there's the intention of updating laradock to use multiple db associate to the multiple sites.

What I'm thinking is that this could be performed within Laravel, but there's a but.

What I have in mind is to create a initial migration that will create the DB if not exists i.e.:

`
class CreateSchemaApp01 extends Migration
{
    /**
     * Run the migrations.
     * Setup App specific DB
     * @return void
     */
    public function up()
    {
        DB::statement('create database IF NOT EXISTS '.env('DB_DATABASE'));
    }

}
`

This would solve all the problems if ever I could preventively set up in my .env file:

`DB_DATABASE=testqueue`

But by doing so, I'll get a Cn't connect error because laravel performing 

`php artisan migrate`

will attempt to perform a specific connection to the selected DB instead of doing a generic connection to the host.
 
Basically to create a  DB, I need a migration , that will need a DB_NAME for connection, but the DB_NAME has to be created.

Any idea for a workaround?

Until now I create manually the DB by connection to Mysql container and then run my migrations.
Obviously this makes to sense for the create db migration to exist.

Otherwise I run just the create db migration having 

`DB_DATABASE=homestead`

then after creation

`DB_DATABASE=testqueue` 

but this is not smooth...

thanks for any advice

**Comment by Mahmoudz** [comment]
@naretini Thanks.
This could work but I don't think it's the best solution! since sometimes you need different mysql user for your Apps.. I think a better way would be to create the db by getting in the mysql container manually. But the best way is to run some sql statements that create custom DB's while container is booting and read those DB's credentials from the docker-compose file. I don't know what other options are there!

**Comment by naretini** [comment]
Well, since Docker is a development env. I don't feel the need of having multiple mysql users (at the moment). Mysql user for me could be always the "homestead".
I agree with you with solution of running a sql script after building the mysql container. I struggled to figure out how to get it working still with no success.

I didn't find yet a comprehensive guide on how to fire a shell script to the mysql container that contains instructions for running sql.

What I found around and did:

- created myscript.sql in /laradock/mysql/files/ :
Contents of myscript.sql are:

create database test01;
GRANT ALL ON *.* TO homestead@'%';

- created init_db.sh :
Contents of bash are

#!/bin/bash
/usr/bin/mysqld_safe &
sleep 5
mysql -uroot -proot < /tmp/myscript.sql

- EDIT /mysql/Dockerfile  adding lines: 
ADD files/myscript.sql /tmp/
ADD files/init_db.sh /tmp/

- EDIT /laradock/docker-compose.yml  (Here I Think I'm missing the HOW-TO)
On Mysql container section:
mysql:
        build: ./mysql
        volumes:
            - mysql:/var/lib/mysql
        ports:
            - "3306:3306"
        environment:
            MYSQL_DATABASE: homestead
            MYSQL_USER: homestead
            MYSQL_PASSWORD: secret
            MYSQL_ROOT_PASSWORD: root
        command: /tmp/init_db.sh

no success....

**Comment by Exadra37** [comment]
I think that proposals #424 and #425 may help here once accepted and implemented.

**Comment by philtrep** [comment]
@naretini Take a look at #404, I think it might help you out.

@Exadra37 We'll have to take a look at the different services for possible conflicts when using the same image multiple times. I look forward to your PR, it should solve quite a few issues we've been having...

**Comment by Exadra37** [comment]
@philtrep using the same image multiple times should not cause any issue, but building a container from it using same ports will cause conflicts for sure, unless I have misunderstood something.

**Comment by philtrep** [comment]
@Exadra37 yes that's what I meant. For example we would need to parametrize the `my.cnf` file since each instance of mysql should bind to a different port.

I'm really looking forward to your solution, I haven't seen any Docker-Compose project allowing for easy duplication + configuration of services yet.

**Comment by Exadra37** [comment]
I will do my best.. still learning Docker ;)
